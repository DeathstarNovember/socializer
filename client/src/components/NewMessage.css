.new-message .form-group {
  margin-bottom: 0px;
}

@media (max-device-width: 767px) {
  .card.new-message {
    border-radius: 0px;
    border-left: none;
    border-right: none;
    border-bottom: none;
  }
}
