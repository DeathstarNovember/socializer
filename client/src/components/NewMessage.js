import React, { useState } from "react";
import { useMutation } from "@apollo/react-hooks";
import gql from "graphql-tag";
import { Card, Form } from "react-bootstrap";
import "./NewMessage.css";

export const CREATE_MESSAGE = gql`
  mutation CreateMessage($conversationId: String!, $body: String!) {
    createMessage(conversationId: $conversationId, body: $body) {
      id
    }
  }
`;

const NewMessage = ({ conversationId }) => {
  const [body, setBody] = useState("");

  const [submit] = useMutation(CREATE_MESSAGE, {
    onCompleted: () => setBody(""),
  });

  return (
    <Card className="new-message mt-2">
      <Card.Body>
        <Form
          data-testid="new-message"
          onSubmit={(e) => {
            e.preventDefault();
            submit({
              variables: { body, conversationId },
            });
          }}
        >
          <Form.Group>
            <Form.Control
              rows="3"
              placeholder="What's on your mind?"
              value={body}
              onChange={(e) => setBody(e.target.value)}
            />
          </Form.Group>
        </Form>
      </Card.Body>
    </Card>
  );
};

export default NewMessage;
