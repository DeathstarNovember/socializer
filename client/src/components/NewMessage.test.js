import React from "react";
import { render, fireEvent, wait } from "@testing-library/react";
import { MockedProvider } from "@apollo/react-testing";
import NewMessage, { CREATE_MESSAGE } from "./NewMessage";

describe("NewMessage", () => {
  it("renders correctly", () => {
    const { container } = render(
      <MockedProvider mocks={[]} addTypename={false}>
        <NewMessage conversationId={123} />
      </MockedProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  it("creates new message", async () => {
    const mocks = [
      {
        request: {
          query: CREATE_MESSAGE,
          variables: { conversationId: 123, body: "Hi there" },
        },
        result: { data: { createMessage: { id: 123 } } },
      },
    ];
    const { getByPlaceholderText, getByTestId } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <NewMessage conversationId={123} />
      </MockedProvider>,
    );
    fireEvent.change(getByPlaceholderText("What's on your mind?"), {
      target: { value: "Hi there" },
    });
    fireEvent.submit(getByTestId("new-message"));
    await wait(() =>
      expect(getByPlaceholderText("What's on your mind?").value).toEqual(""),
    );
  });
});
