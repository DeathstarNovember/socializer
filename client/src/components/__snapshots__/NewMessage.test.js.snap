// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewMessage renders correctly 1`] = `
<div>
  <div
    class="new-message mt-2 card"
  >
    <div
      class="card-body"
    >
      <form
        class=""
        data-testid="new-message"
      >
        <div
          class="form-group"
        >
          <input
            class="form-control"
            placeholder="What's on your mind?"
            rows="3"
            value=""
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
